### PR TITLE
MRG: _find_matching_sidecar() should respect passed BIDSPath's suffix and extension

### PR DESCRIPTION
**Pull request by hoechenberger**
I assume I introduced a bug here via #878 which just bit us in the BIDS-Pipeline CI 😓

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [ ] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [ ] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -97,7 +97,7 @@ Bug fixes
 
 - :func:`mne_bids.write_raw_bids` now raises an exception if the provided :class:`mne_bids.BIDSPath` doesn't contain ``subject`` and ``task`` entities, which are required for neurophysiological data, by `Richard Höchenberger`_ (:gh:`903`)
 
-- :func:`mne_bids.read_raw_bids` now handles datasets with multiple electrophysiological data types correctly, by `Richard Höchenberger`_ (:gh:`910`)
+- :func:`mne_bids.read_raw_bids` now handles datasets with multiple electrophysiological data types correctly, by `Richard Höchenberger`_ (:gh:`910`, :gh`915`)
```
Comment (hoechenberger) [inline doc/whats_new.rst:100]:
```suggestion
- :func:`mne_bids.read_raw_bids` now handles datasets with multiple electrophysiological data types correctly, by `Richard Höchenberger`_ (:gh:`910`, :gh`916`)
```

**Review by agramfort — APPROVED** [review]
merge if happy although there is no test. If you can...

**Comment by hoechenberger** [comment]
I'll try to cook up some test cases, good point.

**Comment by hoechenberger** [comment]
All green now, 100% test coverage, merging.

**Pull request merged**